src/js/PhotoSphereViewer.public.js

```diff
@@ -487,3 +487,45 @@ PhotoSphereViewer.prototype.startKeyboardControl = function() {
 PhotoSphereViewer.prototype.stopKeyboardControl = function() {
   window.removeEventListener('keydown', this);
 };
+
+/**
+ * Preload a panorama image and save it into internal cache.
+ * @param {String} pano - the file path
+ * @param {Function} callback - Progress callback, will receive the percentage as argument.
+ */
+PhotoSphereViewer.prototype.preloadPano = function(pano, callback) {
```

Comment:
Not that I don't like, but I want to keep to API clear and consistent. As promise/A+ spec does not include a progress event and this particular usage is not critical, I prefer to remove it.